### PR TITLE
[KED-1315] Add secret scan CircleCI step

### PR DESCRIPTION
**Pull request by ZainPatelQB**
## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

## Motivation and Context
This PR was created to add in a secret scanning step into our CI, using trufflehog to detect any high entropy strings that may acidentally find their way into the repo.

The CircleCI step simply runs trufflehog, ignoring certain files that have high entropy strings (READM/RELEASE/npm package json files) and only runs of the diff of the change (to keep it fast).

## How has this been tested?
Manually tested locally and verified it works by looking in CircleCI.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes
- [x] Assigned myself to the PR
- [x] Added `Type` label to the PR

### CODE REVIEW

File: trufflehog-ignore.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package.json
+package-lock.json
+RELEASE.md
```
Comment (921kiyo) [inline trufflehog-ignore.txt]: Why are release note and readme being ignored?

File: Makefile [hunk 2]
```diff
@@ -30,6 +30,9 @@ pylint:
 	pylint -j 0 --disable=missing-docstring,no-name-in-module package/features
 	flake8 package
 
+secret-scan:
+	trufflehog --max_depth 1 --exclude_path trufflehog-ignore.txt .
```
Comment (921kiyo) [inline Makefile]: `--regex` flag doesn't work in this repo?

File: trufflehog-ignore.txt (hunk 1 above)
Comment (ZainPatelQB) [reply trufflehog-ignore.txt]: Because they contain things like github.com/quantumblacklabs/blob/<high-entropy-string> :(

**Comment by richardwestenra** [comment]
Looks good, but do you mind editing the PR description to add some context, an explanation of what you did and why, and any relevant development notes?

**Comment by ZainPatelQB** [comment]
> Looks good, but do you mind editing the PR description to add some context, an explanation of what you did and why, and any relevant development notes?

Done!

**Comment by richardwestenra** [comment]
@ZainPatelQB Hey we've got a problem with the GPL license on trufflehog: 
https://app.snyk.io/org/quantumblack-labs/project/a15398fb-95bf-4d80-b73d-ff58e45f2b50/pr-check/6dab5110-275f-4787-850b-206aa7163cb6/license?refs=5e862882007f6f5cc310a785e1bf73e4bd7f6d22..33bccdc4c9bc845e14825d7c9f42aa45081ca066

image

**Comment by ZainPatelQB** [comment]
Yep @richardwestenra I'm waiting on approval to ignore the license before merging it in!

**Comment by DanSymons** [comment]
Hey - just re-visiting this to approve the acceptance. From memory, the license errors don't matter here because Trufflehog is only used in dev and it's not a dependency for your OS code, right? Happy to discuss further if I am wrong?

**Comment by ZainPatelQB** [comment]
That’s the correct rationale afaik, thanks Dan!

**Comment by DanSymons** [comment]
Approval from me for you to suppress these - and thank you for looping me in. I would love to sit with someone soon to draft a brief guideline about empowering you to potentially exercise discretion in the suppression of similar errors in dev-only dependencies. I will reach out when back later this month :)

**Comment by ZainPatelQB** [comment]
That sounds super, thank you! You know where to find me :tada:

**Pull request merged**